front/app/components/EsriMap/utils.tsx

```diff
@@ -33,8 +35,62 @@ import {
   DEFAULT_TILE_PROVIDER,
   MAPTILER_ATTRIBUTION,
 } from './constants';
+import messages from './messages';
 import { DefaultBasemapType } from './types';
 
+// For Accessibility: add (aria-labels) to the expand buttons for layer list and legend
+interface UseLabelExpandButtonsProps {
+  mapView: MapView | null;
+  formatMessage: (messages: { id: string; defaultMessage: string }) => string;
+}
+export const useLabelExpandButtons = ({
+  mapView,
+  formatMessage,
+}: UseLabelExpandButtonsProps) => {
+  useEffect(() => {
```

Comment:
You could also use `const { formatMessage } = useIntl()` here, so you have to pass less props around.